src/makeInfinityAdder.js

```diff
@@ -6,7 +6,33 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (a) => {
+    const sum = (b) => {
+      if (b >= 0) {
+        result += b;
+
+        return sum;
+      }
+
+      const temp = result;
+
+      result = 0;
+
+      return temp;
+    };
+
+    if (a) {
+      result += a;
+
+      return sum;
+    }
+
+    return result;
+  };
```

Comment:
let simplify a bit
Check if there is an argument - if so - we add to result, if not we reset result and return value

```suggestion
  const adder = (a) => {
    if (a === undefined) {
      const total = result;
      result = 0;
      
      return total;
    }
    
    result += a;
    
    return adder;
  };
```